From 11c4a48bea0301c0c5b015cee2007404a900a87b Mon Sep 17 00:00:00 2001
From: Ciprian Costea <ciprianmarian.costea@nxp.com>
Date: Thu, 19 May 2022 13:34:39 +0300
Subject: [PATCH 06/30] s32-gen1: cpu: Add 'nvmem-cell' properties to secondary
 cores

This is required in order for 'CPU_CMD' commands to
work properly

Issue: ALB-8901
Upstream-Status: Pending 

Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/s32cc.dtsi | 15 +++++++++++++++
 arch/arm/dts/s32g3.dtsi | 20 ++++++++++++++++++++
 2 files changed, 35 insertions(+)

diff --git a/arch/arm/dts/s32cc.dtsi b/arch/arm/dts/s32cc.dtsi
index 9acf8a3c19..4c2564e243 100644
--- a/arch/arm/dts/s32cc.dtsi
+++ b/arch/arm/dts/s32cc.dtsi
@@ -109,18 +109,33 @@
 			device_type = "cpu";
 			compatible = "arm,cortex-a53";
 			reg = <0x1>;
+
+			nvmem-cells = <&soc_letter>, <&part_no>,
+			    <&soc_major>, <&soc_minor>, <&soc_subminor>;
+			nvmem-cell-names = "soc_letter", "part_no",
+			    "soc_major", "soc_minor", "soc_subminor";
 		};
 
 		cpu2: cpu@100 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a53";
 			reg = <0x100>;
+
+			nvmem-cells = <&soc_letter>, <&part_no>,
+			    <&soc_major>, <&soc_minor>, <&soc_subminor>;
+			nvmem-cell-names = "soc_letter", "part_no",
+			    "soc_major", "soc_minor", "soc_subminor";
 		};
 
 		cpu3: cpu@101 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a53";
 			reg = <0x101>;
+
+			nvmem-cells = <&soc_letter>, <&part_no>,
+			    <&soc_major>, <&soc_minor>, <&soc_subminor>;
+			nvmem-cell-names = "soc_letter", "part_no",
+			    "soc_major", "soc_minor", "soc_subminor";
 		};
 	};
 
diff --git a/arch/arm/dts/s32g3.dtsi b/arch/arm/dts/s32g3.dtsi
index 67027b14d1..3409f7fdd4 100644
--- a/arch/arm/dts/s32g3.dtsi
+++ b/arch/arm/dts/s32g3.dtsi
@@ -49,24 +49,44 @@
 		device_type = "cpu";
 		compatible = "arm,cortex-a53";
 		reg = <0x2>;
+
+		nvmem-cells = <&soc_letter>, <&part_no>,
+			<&soc_major>, <&soc_minor>, <&soc_subminor>;
+		nvmem-cell-names = "soc_letter", "part_no",
+			"soc_major", "soc_minor", "soc_subminor";
 	};
 
 	cpu5: cpu@3 {
 		device_type = "cpu";
 		compatible = "arm,cortex-a53";
 		reg = <0x3>;
+
+		nvmem-cells = <&soc_letter>, <&part_no>,
+			<&soc_major>, <&soc_minor>, <&soc_subminor>;
+		nvmem-cell-names = "soc_letter", "part_no",
+			"soc_major", "soc_minor", "soc_subminor";
 	};
 
 	cpu6: cpu@102 {
 		device_type = "cpu";
 		compatible = "arm,cortex-a53";
 		reg = <0x102>;
+
+		nvmem-cells = <&soc_letter>, <&part_no>,
+			<&soc_major>, <&soc_minor>, <&soc_subminor>;
+		nvmem-cell-names = "soc_letter", "part_no",
+			"soc_major", "soc_minor", "soc_subminor";
 	};
 
 	cpu7: cpu@103 {
 		device_type = "cpu";
 		compatible = "arm,cortex-a53";
 		reg = <0x103>;
+
+		nvmem-cells = <&soc_letter>, <&part_no>,
+			<&soc_major>, <&soc_minor>, <&soc_subminor>;
+		nvmem-cell-names = "soc_letter", "part_no",
+			"soc_major", "soc_minor", "soc_subminor";
 	};
 };
 
-- 
2.17.1

